src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1327,38 +1306,47 @@ bool NetworkOPsImp::checkLastClosedLedger (
             ++ledgers[peerLedger].nodesUsing;
     }
 
-    auto bestVC = ledgers[closedLedger];
 
     // 3) Is there a network ledger we'd like to switch to? If so, do we have
     // it?
     bool switchLedgers = false;
+    ValidationCount bestCounts = ledgers[closedLedger];
 
     for (auto const& it: ledgers)
```

Comment:
I think that this entire loop (and a bit of the surrounding code) can be simplified to this:

```
        // Ensure that an entry for the closed ledger always exists.
        ledgers.insert ({ closedLedger, {} });

        // Find the best entry
	auto result = std::max_element(ledgers.begin(), ledgers.end(),
		[](auto const& best, auto const& curr)
		{
			// Prefer ledger with more trustedValidations
			if (curr.second.trustedValidations > best.second.trustedValidations)
			    return true;
			if (curr.second.trustedValidations < best.second.trustedValidations)
			    return false;
			// If neither are trusted, prefer more nodesUsing
			if (curr.second.trustedValidations == 0)
			{
			    if (curr.second.nodesUsing > best.second.nodesUsing)
			        return true;
			    if (curr.second.nodesUsing < best.second.nodesUsing)
			        return false;
			}
			// Resolve ties by preferring the higher ledger hash
			return curr.first > best.first;
		});
```

At this point, `result` points to an entry in the `ledgers` map that is the best. We can determine if we need to switch, simply by comparing `result->first` with `closedLedger`.